django: Cast to CharField with max_length doesn't work

```
======================================================================
FAIL: test_cast_to_char_field_with_max_length (db_functions.comparison.test_cast.CastTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/django/test/utils.py", line 373, in inner
    return func(*args, **kwargs)
  File "/home/tim/code/django/django/test/testcases.py", line 1229, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/home/tim/code/django/tests/db_functions/comparison/test_cast.py", line 38, in test_cast_to_char_field_with_max_length
    self.assertEqual(names.get().cast_string, 'B')
AssertionError: 'Bob' != 'B'
- Bob
+ B
```

--- a/spanner/django/features.py
+++ b/spanner/django/features.py
@@ -196,9 +196,6 @@ class DatabaseFeatures(BaseDatabaseFeatures):
         # null: https://github.com/orijtech/spanner-orm/issues/242
         'db_functions.text.test_concat.ConcatTests.test_basic',
         'db_functions.text.test_concat.ConcatTests.test_many',
-        # Cast to CharField with max_length doesn't work:
-        # https://github.com/orijtech/spanner-orm/issues/258
-        'db_functions.comparison.test_cast.CastTests.test_cast_to_char_field_with_max_length',
         # casting DateField to DateTimeField adds an unexpected hour:
         # https://github.com/orijtech/spanner-orm/issues/260
         'db_functions.comparison.test_cast.CastTests.test_cast_from_db_date_to_datetime',

--- a/spanner/django/functions.py
+++ b/spanner/django/functions.py
@@ -7,8 +7,19 @@
 import math
 
 from django.db.models.functions import (
-    Chr, Cot, Degrees, Left, Log, Ord, Pi, Radians, Right, StrIndex, Substr,
+    Cast, Chr, Cot, Degrees, Left, Log, Ord, Pi, Radians, Right, StrIndex,
+    Substr,
 )
+
+
+def cast(self, compiler, connection, **extra_context):
+    # Account for a field's max_length using SUBSTR.
+    max_length = getattr(self.output_field, 'max_length', None)
+    if max_length is not None:
+        template = 'SUBSTR(' + self.template + ', 0, %s)' % max_length
+    else:
+        template = self.template
+    return self.as_sql(compiler, connection, template=template, **extra_context)
 
 
 def chr_(self, compiler, connection, **extra_context):
@@ -64,6 +75,7 @@ def substr(self, compiler, connection, **extra_context):
 
 
 def register_functions():
+    Cast.as_spanner = cast
     Chr.as_spanner = chr_
     Cot.as_spanner = cot
     Degrees.as_spanner = degrees

